docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.list_ops.ListOp`
+     - No direct replacement. This is the base class for operator lists. In general, these could be replaced with
+       Python ``list``\s. For ``Pauli`` operators, there are a few alternatives, depending on the use-case.
+       One alternative is :class:`~qiskit.quantum_info.PauliList`.
+
+   * - :class:`~qiskit.opflow.list_ops.ComposedOp`
+     - No direct replacement. Current workflows do not require composition of states and operators within
+       one object (no lazy evaluation).
+
+   * - :class:`~qiskit.opflow.list_ops.SummedOp`
+     - No direct replacement. For ``Pauli`` operators, use :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+   * - :class:`~qiskit.opflow.list_ops.TensoredOp`
+     - No direct replacement. For ``Pauli`` operators, use :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+
+State Functions
+---------------
+*Back to* `Index`_
+
+The :mod:`~qiskit.opflow.state_fns` module can be generally replaced by :class:`~qiskit.quantum_info.QuantumState`,
+with some differences to keep in mind:
+
+1. The primitives-based workflow does not rely on constructing state functions as opflow did
+2. Algorithm-specific functionality has been migrated to the respective algorithm's module
+
+
+Similarly to :class:`~qiskit.opflow.primitive_ops.PrimitiveOp`, :class:`~qiskit.opflow.state_fns.StateFn`
+acts as a factory to create the corresponding sub-class depending on the computational primitive used to initialize it.
+
+.. tip::
+
+    Interpreting :class:`~qiskit.opflow.state_fns.StateFn` as a factory class:
+
+    .. list-table::
+       :header-rows: 1
+
+       * - Class passed to :class:`~qiskit.opflow.state_fns.StateFn`
+         - Sub-class returned
+
+       * - ``str``, ``dict``, :class:`~qiskit.result.Result`
+         - :class:`~qiskit.opflow.state_fns.DictStateFn`
+
+       * - ``list``, ``np.ndarray``, :class:`~qiskit.quantum_info.Statevector`
+         - :class:`~qiskit.opflow.state_fns.VectorStateFn`
+
+       * - :class:`~qiskit.circuit.QuantumCircuit`, :class:`~qiskit.circuit.Instruction`
+         - :class:`~qiskit.opflow.state_fns.CircuitStateFn`
+
+       * - :class:`~qiskit.opflow.OperatorBase`
+         - :class:`~qiskit.opflow.state_fns.OperatorStateFn`
+
+This means that references to :class:`~qiskit.opflow.state_fns.StateFn` in opflow code should be examined to
+identify the sub-class that is being used, to then look for an alternative.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.state_fns.StateFn`
+     - In most cases, :class:`~qiskit.quantum_info.Statevector`. Remember that this is a factory class.
+
+   * - :class:`~qiskit.opflow.state_fns.CircuitStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.DictStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
```

Comment:
The ``DictStateFn`` could store an efficient representation if the measurement results were sparse, which the ``Statevector`` cannot, so I think a more accurate replacement would be the ``qiskit.result.QuasiDist`` or the ``Result`` class you mentioned above 🙂

Reply:
Yes, I forgot to update this part of the table, thanks for catching it!